core/src/main/scala/akka/persistence/query/EventBySliceEnvelope.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.query
+
+import java.util.Optional
+
+import akka.util.HashCode
+
+object EventBySliceEnvelope {
```

Comment:
Do we need to add Slice in the name? 

In an ideal worlds, we should have one kind of envelope that is typed and supporting lazy loading. 
If we have: `akka.persistence.query.typed.EventEnvelope` we could gradually move all queries to it (with proper deprecation cycle) and also AkkaProjection. 

We don't need to impose us any deadline about it, but we can leave the door open by using a more generic name.

Reply:
> Do we need to add Slice in the name?

Yeah, I had to come up with a different name, but I can use a different package name as you suggest.

Then we should place the new query interfaces in that `typed` package as well.

Reply:
good idea to also put the queries under `typed`. That gives us a migration and deprecation path for the existing queries.

Reply:
done in https://github.com/akka/akka-persistence-r2dbc/pull/144/commits/f900a16926c74b682edabfd58c8490b978b58c38